Unable to build/run tests with cmake on RHEL 7/8

I have been using this command to test that runtime's main branch can build and run tests on RHEL 7 and CMake 3.6.2:

```
./build.sh --subset clr+libs+libs.tests --test --runtimeconfiguration Release --librariesConfiguration Debug /p:WithoutCategories=IgnoreForCI /p:NoPgoOptimize=true
```

Some time between April 4 (commit 21490fc0f3aa5c3c99c2bda8f4e36fd3f7233950) and April 6 (commit bc8be66f0e0aa119c5c16599de88326ab8e5bc5c), this command regressed. Now it fails to build:

```
  -- Performing Test HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES
  -- Performing Test HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES - Success
  -- Found PythonInterp: /usr/bin/python (found version "2.7.5") 
  CMake Error at pal/tests/CMakeLists.txt:1 (cmake_minimum_required):
    CMake 3.14.5 or higher is required.  You are running version 3.6.2
  
  
  -- Configuring incomplete, errors occurred!
  See also "/home/tester/runtime/artifacts/obj/coreclr/Linux.x64.Release/CMakeFiles/CMakeOutput.log".
  See also "/home/tester/runtime/artifacts/obj/coreclr/Linux.x64.Release/CMakeFiles/CMakeError.log".
  ~/runtime/src/coreclr
  ~/runtime/artifacts/obj/coreclr/Linux.x64.Release ~/runtime/src/coreclr
  Executing make install -j 4
  make: *** No rule to make target `install'.  Stop.
  ~/runtime/src/coreclr
  Failed to build "CoreCLR component".
```

Is the change that broke this an accident or is this an intentional? Is building/running tests on cmake 3.6.2 a supported configuration? 

cc @dleeapho  @tmds

I am also hitting this on RHEL 8:

```
  -- Performing Test HAVE_FULLY_FEATURED_PTHREAD_MUTEXES
  -- Performing Test HAVE_FULLY_FEATURED_PTHREAD_MUTEXES - Success
  -- Performing Test HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES
  -- Performing Test HAVE_FUNCTIONAL_PTHREAD_ROBUST_MUTEXES - Success
  -- Found PythonInterp: /usr/bin/python3.6 (found version "3.6.8") 
  CMake Error at pal/tests/CMakeLists.txt:1 (cmake_minimum_required):
    CMake 3.14.5 or higher is required.  You are running version 3.11.4
  
  
  -- Configuring incomplete, errors occurred!
  See also "/home/tester/runtime/artifacts/obj/coreclr/Linux.x64.Release/CMakeFiles/CMakeOutput.log".
  See also "/home/tester/runtime/artifacts/obj/coreclr/Linux.x64.Release/CMakeFiles/CMakeError.log".
  ~/runtime/src/coreclr
  ~/runtime/artifacts/obj/coreclr/Linux.x64.Release ~/runtime/src/coreclr
  Executing make install -j 4
  make: *** No rule to make target 'install'.  Stop.
  ~/runtime/src/coreclr
  Failed to build "CoreCLR component".
```

--- a/src/coreclr/pal/tests/CMakeLists.txt
+++ b/src/coreclr/pal/tests/CMakeLists.txt
@@ -1,5 +1,3 @@
-cmake_minimum_required(VERSION 3.14.5)
-
 # Compile options
 add_definitions(-DLP64COMPATIBLE)
 add_definitions(-DCORECLR)

